src/components/pages/DetailsPage/DetailsPage.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import style from './DetailsPage.module.scss';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ROUTER_PATHS, URL } from '../../../models/enums';
+import { ShowData } from '../../../models/interfaces';
+import { AiFillStar } from 'react-icons/ai';
+import { GoLinkExternal } from 'react-icons/go';
+import Spinner from '../../Spinner/Spinner';
+
+interface DetailsPageProps {
+  setIsCardItemsDarked: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const DetailsPage: React.FC<DetailsPageProps> = ({ setIsCardItemsDarked }) => {
+  const { id } = useParams();
+
+  const navigate = useNavigate();
+
+  const goBack = () => {
+    navigate(`${ROUTER_PATHS.MAIN}`);
+    setIsCardItemsDarked(false);
+  };
+
+  const noDate = 'no data to show';
+
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [show, setShow] = React.useState({
+    id: 0,
+    image: '',
+    name: '',
+    rating: 0,
+    language: '',
+    summary: '',
+    genres: [noDate],
+    premiered: '',
+    ended: '',
+    officialSite: '',
+  });
+
+  const getShowById = async () => {
+    if (isLoading) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      const response = await fetch(`${URL.SHOWS}/${id}`);
+
+      if (!response.ok) {
+        throw new Error('Network response error');
+      }
+
+      const data: ShowData = await response.json();
+
+      show.genres = [];
```

Comment:
Why do you change the state directly?

Reply:
Because sometimes `show.genres` is equal `null` and I need to render something instead. What is the best way to do it?